### PR TITLE
Fixup Z-Perl to work with the Classic Burning Crusade.

### PR DESCRIPTION
**Pull request by glajchs**
Set the TOC for Burning Crusade Classic.  Fix the version checking code within the addon to work for BC as well as vanilla classic.  Remove link to the lib classic durations embedded library, as it wasn't working in BC.

### CODE REVIEW

File: ZPerl/Libs/LibClassicCasterino/LibClassicCasterino.lua [hunk 1]
```diff
@@ -2,7 +2,7 @@
 LibClassicCasterino
 Author: d87
 --]================]
-if WOW_PROJECT_ID ~= WOW_PROJECT_CLASSIC then return end
+if WOW_PROJECT_ID >= WOW_PROJECT_CLASSIC then return end
```
Comment (corporategoth) [inline ZPerl/Libs/LibClassicCasterino/LibClassicCasterino.lua:5]: Wrong logic.  This is supposed to return if not classic.  You just made it return IF classic.

I would go with
```
if WOW_PROJECT_ID == WOW_PROJECT_MAINLINE then return end
```

File: ZPerl/Libs/LibClassicDurations/core.lua [hunk 2]
```diff
@@ -17,7 +17,7 @@ Usage example 1:
 	end
 
 --]================]
-if WOW_PROJECT_ID ~= WOW_PROJECT_CLASSIC then return end
+if WOW_PROJECT_ID >= WOW_PROJECT_CLASSIC then return end
```
Comment (corporategoth) [inline ZPerl/Libs/LibClassicDurations/core.lua:20]: Wrong logic.  This is supposed to return if not classic.  You just made it return IF classic.

I would go with
```
if WOW_PROJECT_ID == WOW_PROJECT_MAINLINE then return end
```

However, if this is a classic ONLY thing, then return it to the original ~= WOW_PROJECTCLASSIC (ie. I saw you removed the usage of this later in ZPerl_Target/ZPerl_Target.lua)

File: ZPerl/ZPerl.lua [hunk 3]
```diff
@@ -11,7 +11,7 @@ XPerl_RequestConfig(function(New)
 end, "$Revision: @project-revision@ $")
 XPerl_SetModuleRevision("$Revision: @project-revision@ $")
 
-local IsClassic = WOW_PROJECT_ID == WOW_PROJECT_CLASSIC
+local IsClassic = WOW_PROJECT_ID >= WOW_PROJECT_CLASSIC
```
Comment (corporategoth) [inline ZPerl/ZPerl.lua:14]: Could also use:

```
local IsClassic = (WOW_PROJECT_ID ~= WOW_PROJECT_MAINLINE)
```

File: ZPerl_Options/ZPerl_FrameOptions.xml [hunk 4]
```diff
@@ -1271,7 +1271,7 @@
 					<OnLoad>
 						self:SetID(self:GetParent():GetID())
 						self.tooltipText = XPERL_CONF_RAID_CLASS_DOWN
-						if (self:GetID() == (WOW_PROJECT_ID == WOW_PROJECT_CLASSIC and 9 or 12)) then
+						if (self:GetID() == (WOW_PROJECT_ID >= WOW_PROJECT_CLASSIC and 9 or 12)) then
```
Comment (corporategoth) [inline ZPerl_Options/ZPerl_FrameOptions.xml:1274]: Ugh, this file is screaming out for an IsClassic variable

File: ZPerl_Target/ZPerl_Target.lua [hunk 5]
```diff
@@ -25,15 +25,15 @@ XPerl_RequestConfig(function(new)
...
-local LCD = IsClassic and LibStub and LibStub("LibClassicDurations")
-if IsClassic then
-	LCD.RegisterCallback("ZPerl", "UNIT_BUFF", function(event, unit)
-		if unit == "target" then
-			XPerl_Target_Events:UNIT_AURA(event, unit)
-		end
-	end)
-end
+local IsClassic = WOW_PROJECT_ID >= WOW_PROJECT_CLASSIC
+--local LCD = IsClassic and LibStub and LibStub("LibClassicDurations")
```
Comment (corporategoth) [inline ZPerl_Target/ZPerl_Target.lua:29]: Why remove this?

**Comment by corporategoth** [comment]
Additionally, I had to make some changes to LibClassicCasterino and LibClassicDuration.

LibClassicDuration has been updated, so getting a new version (and fixing the classic trigger) works fine for that.
LibClassicCasterino I had to modify to check if GetSpellInfo(spellid) returned nil in several places.

After these updates, as well as the above, no more errors ;)

**Comment by Resike** [comment]
Thanks boys, but i got this.

**Pull request closed without merging**